### DISCUSSION TITLE
Random decompression errors

### DISCUSSION THREAD

**Discussion by Polar-Pumpkin** [rikyoz/bit7z#182] [Q&A] [answered]
I may encounter "Cannot read file entry: 参数错误。" (from ex.what()) when decompressing on other computer. Trying to get the HRESULT value, the print shows: `-2147024809`, but I can't find the corresponding HRESULT meaning in my search. 😢 

But I use NUnit on my own computer, call my C++ DLL from C#, and decompress it 10,000 times without encountering an error.

**Comment by rikyoz** [comment, marked as answer]
The `-2147024809` error code corresponds to the `E_INVALIDARG` HRESULT error (`0x80070057` in hex, https://learn.microsoft.com/en-us/windows/win32/seccrypto/common-hresult-values).
This same error happens due to the issue https://github.com/rikyoz/bit7z/issues/181, so you may be experiencing it too.
If this is the case, the problem is due to an old bug in MSVC's std::filesystem, which was fixed only in the latest versions of the compiler.
One possible workaround that I proposed is to use the new `BIT7Z_DISABLE_USE_STD_FILESYSTEM` CMake option provided in the `hotfix/v4.0.5` branch, as it makes bit7z avoid using std::filesystem altogether.

**Reply by Polar-Pumpkin** [reply]
Thank you! I'll try it 🥰

**Reply by Polar-Pumpkin** [reply]
After using `hotfix/v4.0.5` I haven't had this problem again so far. <3

**Reply by Polar-Pumpkin** [reply]
I'm really curious about how you found the corresponding HEX values. I tried using some online conversion tools, but the HEX values I got don't seem correct. 😢

**Reply by Polar-Pumpkin** [reply]
I just asked ChatGPT, and it helped me convert to a HEX value using the following code.

```python
# Converting the given integer to its corresponding hexadecimal representation

integer_value = -2147417850
hex_value = hex(integer_value & 0xFFFFFFFF)  # Using bitwise AND for two's complement representation
hex_value # 0x80010106
```

**Reply by rikyoz** [reply]
> I'm really curious about how you found the corresponding HEX values. I tried using some online conversion tools, but the HEX values I got don't seem correct. 😢

I actually used the Windows calculator app in "Programmer" mode. I just pasted the `-2147024809` as a decimal value and then checked the corresponding hex value (ignoring the `FFFFFFFF` prefix):

![image](https://github.com/rikyoz/bit7z/assets/1334291/8a43d746-ca6a-4b3d-bc32-d70303040ab0)

This is basically the same operation ChatGPT suggested as Python code.
